src/Microsoft.ML.Transforms/Text/NgramHashingTransformer.cs

```diff
@@ -878,7 +878,8 @@ public sealed class ColumnOptions
             /// <summary>Name of the column resulting from the transformation of <see cref="InputColumnNames"/>.</summary>
             public readonly string Name;
             /// <summary>Names of the columns to transform.</summary>
-            public readonly string[] InputColumnNames;
+            public IReadOnlyList<string> InputColumnNames => InputColumnNamesArray;
```

Comment:
How can someone set these values?

Reply:
You set them in the constructor. This like the other parameters was not intended to be mutable.

Reply:
Obviously this concern existed before your change, but

I think our "Options" classes are inconsistent here.

For example in TextLoader, I can set the properties after the constructor:

https://github.com/dotnet/machinelearning/blob/002d6b45e1fd9f699129a322180e135a20f65483/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs#L101-L129

I would have thought that these Options classes are mutable - just like how TextLoader's Column class is.

Reply:
I agree that there is an inconsistency -- I think this is an effect of something that I observed that some things that had been renamed from `ColumnInfo` to `ColumnOptions` shouldn't have actually been renamed in #2709. But until that quesiton of renaming of the *class* is addressed, I can only make sure the object itself is behaving consistently w.r.t. itself, and ***this*** is very, very clearly intended to be a non-mutable structure.